### DISCUSSION TITLE
Why use bind mount instead of docker volume for database folder?

### DISCUSSION THREAD

**Discussion by helgehatt** [immich-app/immich#14453] [Q&A] [answered]
If I recall correctly, earlier version of docker-compose.yml did not include the `${DB_DATA_LOCATION}` environment variable and instead used a docker volume for `/var/lib/postgresql/data`. Is there any practical reason for this change?

I believe the correct way to backup the database data is using `pg_dump` like Immich does automatically for us now (the dumps saved at `UPLOAD_LOCATION/backups`.

**Comment by alextran1502** [comment, marked as answer]
With the docker volume approach, users can mistakenly type in `docker-compose down -V`, which will remove the volume and hence remove your database.

**Comment by danieldietzler** [comment]
The primary reason to switch was people using `docker compose down -v` because some tutorial said so, and thus accidentally deleting their database

**Reply by danieldietzler** [reply]
Nvm 😅

**Reply by alextran1502** [reply]
ding

**Reply by alextran1502** [reply]
+1 wpm

**Reply by danieldietzler** [reply]
Hahaha :P

**Comment by helgehatt** [comment]
Got it, thanks
